modules/apps/frontend-js/frontend-js-a11y-web/src/main/java/com/liferay/frontend/js/a11y/web/internal/servlet/taglib/A11yBottomJSPDynamicInclude.java

```diff
@@ -143,7 +143,7 @@ public void register(
 
 		if (FFA11yConfigurationUtil.getEnable()) {
 			dynamicIncludeRegistry.register(
-				"/html/common/themes/top_head.jsp#post");
+				"/html/common/themes/bottom.jsp#post");
```

Comment:
We're relying in how browser will process scripts in a given HTML. I think if it's solves temporally it's fine but would be good to have an API for it. 

@dsanz, I think We can include it as an evidence for adding ["allPortletsReady"](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/1666844800/LPS-135643+-+allPortletsReady+event).

Reply:
Well, why wouldn't we trust? the Browser's execution flow is exactly that, first the CSS are downloaded and executed, scripts in the head and then the rendering happens, the scripts are blocking, then the browser blocks rendering when it starts executing a script, HTML and CSS rendering are also blocked, so when we get to our script we will have Senna started because we are talking about the single thread here, so we must trust.

The race condition problem is that we can have A11y being executed before Senna.js, that way we are guaranteeing that Senna.js will always be executed first and after all the head is executed, we will be executing A11y.

Reply:
> so we must trust

We had some issues with the way that scripts are executed in browsers before, depending on the vendor. My idea was just to bring it and say that We must a more reliable way to make sure our script will be just run when everything was already loaded and there aren't any scripts in the queue

Reply:
Hmm, code download is parallel but execution is synchronous so it's safe to trust the browser, it probably won't change because JavaScript is a single thread and the problem we had was in IE 11 but not because it was different but because we handled it ourselves the execution with Senna.js, and the problem was related to CSS scripts, inline vs asynchronous, the inline CSS was executed as soon as we added it to the DOM but the asynchronous CSS would still download the file and then running soon, it just caused ordering problems of CSS declaration that influences the rule.

I say we can trust because we have over 100 scripts and CSS running in the head, so I registered them at the bottom instead of adding them at the head as the last in the queue 😉.